app/gui/view/src/root.rs

```diff
@@ -0,0 +1,152 @@
+//! Root View of the IDE.
+//!
+//! The main entry point to the IDE which can display either Welcome Screen or Project View.
+//! Initially displays Welcome Screen. As soon as user selects a project to open, Root View
+//! switches state to Project View and stays in this state till the application quit.
+// TODO [vitvakatu]: at the moment Root View is not used as described, but it will be used in the
+// following tasks.
+
+use ensogl::prelude::*;
+
+use enso_frp as frp;
+use ensogl::application;
+use ensogl::application::Application;
+use ensogl::display;
+use std::rc::Rc;
+
+
+// =============
+// === Model ===
+// =============
+
+/// Two possible states of Root View.
+#[derive(Clone, Debug, PartialEq)]
+enum State {
+    /// Displaying Welcome Screen.
+    WelcomeScreen,
+    /// Displaying Project View with some opened project.
+    OpenedProject,
+}
+
+/// Root View model. Stores both Welcome Screen and Project View views and handles their
+/// visibility.
+#[derive(Clone, CloneRef, Debug)]
+pub struct Model {
+    app:            Application,
+    logger:         Logger,
+    display_object: display::object::Instance,
+    state:          Rc<CloneCell<State>>,
+    welcome_view:   crate::welcome_screen::View,
+    project_view:   crate::project::View,
+}
+
+impl Model {
+    /// Constuctor. Creates both Welcome Screen and Project View.
+    pub fn new(app: &Application) -> Self {
+        let app = app.clone_ref();
+        let logger = Logger::new("RootView");
+        let display_object = display::object::Instance::new(&logger);
+        let state = Rc::new(CloneCell::new(State::WelcomeScreen));
+        let welcome_view = app.new_view::<crate::welcome_screen::View>();
+        let project_view = app.new_view::<crate::project::View>();
+        display_object.add_child(&welcome_view);
+
+        Self { app, logger, display_object, welcome_view, project_view, state }
+    }
+
+    /// Switch displayed view from Welcome Screen to Project View.
+    pub fn switch_view(&self) {
```

Comment:
The name of the function tells something different than the comment. If this switches things only in one direction, it should be named `switch_welcome_screen_to_project_view`. Everyone should understand what function does by its name. If the function is named `switch_view` then it should switch the view (both sides).